restapi.go

```diff
@@ -2339,3 +2361,215 @@ func (s *Session) RelationshipsMutualGet(userID string) (mf []*User, err error)
+// guildID     : Guild ID to create guild-specific application command. If empty - creates global appication command.
+// cmd         : New application command data.
+func (s *Session) ApplicationCommandCreate(appID string, guildID string, cmd *ApplicationCommand) (ccmd *ApplicationCommand, err error) {
+	if appID == "" {
+		appID = s.State.User.ID
+	}
+
+	endpoint := EndpointApplicationGlobalCommands(appID)
+	if guildID != "" {
+		endpoint = EndpointApplicationGuildCommands(appID, guildID)
+	}
+
+	var body []byte
+	body, err = s.RequestWithBucketID("POST", endpoint, *cmd, endpoint)
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &ccmd)
+
+	return
+}
+
+// ApplicationCommandEdit edits application command and returns old command data.
+// appID       : The application ID. If empty - s.State.User.ID. Note: works only with opened session.
+// cmdID       : Application command ID to edit.
+// guildID     : Guild ID to edit guild-specific application command. If empty - edits global application command.
+// cmd         : Updated application command data.
+func (s *Session) ApplicationCommandEdit(appID, cmdID, guildID string, cmd *ApplicationCommand) (oldcmd *ApplicationCommand, err error) {
+	if cmdID == "" {
+		return nil, errors.New("invalid command id")
+	}
+
+	if appID == "" {
+		appID = s.State.User.ID
+	}
+
+	endpoint := EndpointApplicationGlobalCommand(appID, cmdID)
+	if guildID != "" {
+		endpoint = EndpointApplicationGuildCommand(appID, guildID, cmdID)
+	}
+
+	var body []byte
+	body, err = s.RequestWithBucketID("PATCH", endpoint, *cmd, endpoint)
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &oldcmd)
+
+	return
+}
+
+// ApplicationCommandDelete deletes application command by id.
+// appID       : The application ID. If empty - s.State.User.ID. Note: works only with opened session.
+// cmdID       : Application command ID to delete.
+// guildID     : Guild ID to delete guild-specific application command. If empty - deletes global application command.
+func (s *Session) ApplicationCommandDelete(appID, guildID, cmdID string) error {
+	if cmdID == "" {
+		return errors.New("invalid command id")
+	}
+
+	if appID == "" {
+		appID = s.State.User.ID
+	}
+
+	endpoint := EndpointApplicationGlobalCommand(appID, cmdID)
+	if guildID != "" {
+		endpoint = EndpointApplicationGuildCommand(appID, guildID, cmdID)
+	}
+
+	_, err := s.RequestWithBucketID("DELETE", endpoint, nil, endpoint)
+
+	return err
+}
+
+// ApplicationCommand retrieves an application command by given id.
+// appID       : The application ID. If empty - s.State.User.ID. Note: works only with opened session.
+// cmdID       : Application command ID.
+// guildID     : Guild ID to retrieve guild-specific application command. If empty - retrieves global application command.
+func (s *Session) ApplicationCommand(appID, guildID, cmdID string) (cmd *ApplicationCommand, err error) {
```

Comment:
In different function I see return `*ApplicationCommand` with name like `ccmd, oldcmd, rcmd` and other. Pl don't use named return here

```
ApplicationCommand(appID, guildID, cmdID string) (*ApplicationCommand, error)
```